relayer/app/worker.go

```diff
@@ -190,6 +190,9 @@ func (w *Worker) newCallback(
 	sendBuffer func(context.Context, xchain.Submission) error,
 	msgStreamMapper msgStreamMapper,
 ) cchain.ProviderCallback {
+	var cachedValSetID uint64
```

Comment:
Is this thread safe? We can potentially update these cached values on line [212](https://github.com/omni-network/omni/pull/2233/files#diff-03ecbc560f3649f21a6909ca284e999bc6b04abc4d612de523f6a28eade9cbe3R212) while they are being read during [submission creation  ](https://github.com/omni-network/omni/pull/2233/files#diff-32d83c46ca4a14f4dce235974a632792e6fec239e78e8b9281822cc246830f8cR66)

Reply:
no need for thread safety, cprovider and xprovider callbacks ares single threaded